### PR TITLE
Add validation,resource_usages,texture,in_render_common:* - Part I

### PR DESCRIPTION
**Pull request by Jiawei-Shao**
This PR adds the first part of validation,resource_usages,texture,*:
- in_render_common:subresources_from_same_texture_as_color_attachments

Issue: #905 

<hr>

**Requirements for PR author:**

- [*] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [*] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [*] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by Jiawei-Shao** [comment]
@kainino0x PTAL, thanks!

File: src/webgpu/api/validation/resource_usages/texture/in_render_common.spec.ts [hunk 1]
```diff
@@ -18,4 +18,75 @@ subresources, one for same draw/same pass/different pass, one for visibilities).
...
+class F extends ValidationTest {
+  getColorAttachment(
+    texture: GPUTexture,
+    textureViewDescriptor?: GPUTextureViewDescriptor
+  ): GPURenderPassColorAttachment {
+    const view = texture.createView(textureViewDescriptor);
+
+    return {
+      view,
+      loadValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
```
Comment (austinEng) [inline src/webgpu/api/validation/resource_usages/texture/in_render_common.spec.ts:30]: API changed. instead use:
```suggestion
      clearValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
      loadOp: 'clear',
```

File: src/webgpu/api/validation/resource_usages/texture/in_render_common.spec.ts [hunk 2]
```diff
@@ -18,4 +18,75 @@ subresources, one for same draw/same pass/different pass, one for visibilities).
...
+    };
+  }
+}
+
+export const g = makeTestGroup(F);
+
+// Test that the different subresource of the same texture are allowed to be used as color
+// attachments in same / different render passes, while the same subresource is only allowed to be
+// used as different color attachments in different render passes.
+g.test('subresources_from_same_texture_as_color_attachments')
```
Comment (austinEng) [inline src/webgpu/api/validation/resource_usages/texture/in_render_common.spec.ts:41]: Overall, this test looks good. We should have a plan for how we're going to get coverage for the rest of the cases. See the plan in the TODO at the top of this file. OK with me to land this test though and potentially reorganize in the future if/when we need to parameterize it to handle different combinations of options.

File: src/webgpu/api/validation/resource_usages/texture/in_render_common.spec.ts [hunk 3]
```diff
@@ -18,4 +18,75 @@ subresources, one for same draw/same pass/different pass, one for visibilities).
 import { makeTestGroup } from '../../../../../common/framework/test_group.js';
 import { ValidationTest } from '../../validation_test.js';
 
-export const g = makeTestGroup(ValidationTest);
+class F extends ValidationTest {
```
Comment (austinEng) [inline src/webgpu/api/validation/resource_usages/texture/in_render_common.spec.ts:21]: "F" could be a more descriptive name

**Review by austinEng — APPROVED** [review]

File: src/webgpu/api/validation/resource_usages/texture/in_render_common.spec.ts (hunk 3 above)
Comment (austinEng) [reply src/webgpu/api/validation/resource_usages/texture/in_render_common.spec.ts:21]: Nevermind, I see many other tests use "F". It is ok to keep it.

**Pull request merged**